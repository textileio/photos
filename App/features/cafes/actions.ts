import { createAsyncAction } from 'typesafe-actions'

// Registering with a new cafe requires the cafe URL and string
// With the new SDK, the URL needs to be changed to the peer id
export const registerCafe = createAsyncAction(
  'cafes/REGISTER_CAFE_REQUEST',
  'cafes/REGISTER_CAFE_SUCCESS',
  'cafes/REGISTER_CAFE_ERROR'
)<
  {
    peerId: string
    token: string
    success?: () => void
  },
  string,
  {
    peerId: string
    error: any
  }
>()

// Deregistering from a cafe requires the cafe's peer id
export const deregisterCafe = createAsyncAction(
  'cafes/DEREGISTER_CAFE_REQUEST',
  'cafes/DEREGISTER_CAFE_SUCCESS',
  'cafes/DEREGISTER_CAFE_ERROR'
)<
  {
    id: string
    success?: () => void
  },
  string,
  {
    id: string
    error: any
  }
>()
