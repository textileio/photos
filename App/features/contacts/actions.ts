import { createAction } from 'typesafe-actions'
import { pb } from '@textile/react-native-sdk'
import Contacts from 'react-native-contacts'

export const getContactsRequest = createAction('GET_CONTACTS_REQUEST')

export const getContactsSuccess = createAction('GET_CONTACT_SUCCESS', (resolve) => {
  return (contacts: ReadonlyArray<pb.IContact>) => resolve({ contacts })
})

export const searchRequest = createAction('@contacts/SEARCH_REQUEST', (resolve) => {
  return (searchString: string) => resolve({ searchString })
})

export const searchStarted = createAction('@contacts/SEARCH_STARTED')

export const searchResultTextile = createAction('@contacts/SEARCH_RESULT_TEXTILE', (resolve) => {
  return (result: pb.IContact) => resolve({ result })
})

export const textileSearchComplete = createAction('@contacts/TEXTILE_SEARCH_COMPLETE')

export const searchResultsAddressBook = createAction('@contacts/SEARCH_RESULTS_ADDRESS_BOOK', (resolve) => {
  return (results: ReadonlyArray<Contacts.Contact>) => resolve({ results })
})

export const searchErrorTextile = createAction('@contacts/SEARCH_ERROR_TEXTILE', (resolve) => {
  return (error: any) => resolve({ error })
})

export const searchErrorAddressBook = createAction('@contacts/SEARCH_ERROR_ADDRESS_BOOK', (resolve) => {
  return (error: any) => resolve({ error })
})

export const clearSearch = createAction('@contacts/CLEAR_SEARCH')

export const addContactRequest = createAction('@contacts/ADD_CONTACT_REQUEST', (resolve) => {
  return (contact: pb.IContact) => resolve({ contact })
})

export const addContactSuccess = createAction('@contacts/ADD_CONTACT_SUCCESS', (resolve) => {
  return (contact: pb.IContact) => resolve({ contact })
})

export const addContactError = createAction('@contacts/ADD_CONTACT_ERROR', (resolve) => {
  return (contact: pb.IContact, error: any) => resolve({ contact, error })
})

export const clearAddContact = createAction('@contacts/CLEAR_ADD_CONTACT_ERROR', (resolve) => {
  return (contact: pb.IContact) => resolve({ contact })
})

export const authorInviteRequest = createAction('@contacts/AUTHOR_INVITE_REQUEST', (resolve) => {
  return (contact: Contacts.Contact) => resolve({ contact })
})
