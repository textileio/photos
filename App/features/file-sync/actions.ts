import { createAction, createAsyncAction } from 'typesafe-actions'

export const begin = createAction('file-sync/BEGIN', resolve => {
  return (group: string, id: string) => resolve({ group, id })
})

export const uploadProgress = createAction(
  'file-sync/UPLOAD_PROGRESS',
  resolve => {
    return (id: string, progress: number) => resolve({ id, progress })
  }
)

export const uploadComplete = createAction(
  'file-sync/UPLOAD_COMPLETE',
  resolve => {
    return (id: string, responseCode: string, responseBody: string) =>
      resolve({ id, responseCode, responseBody })
  }
)

export const error = createAction('file-sync/ERROR', resolve => {
  return (id: string, error: string) => resolve({ id, error })
})
