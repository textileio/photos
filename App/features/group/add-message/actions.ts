import { createAsyncAction } from 'typesafe-actions'

export const addMessage = createAsyncAction(
  '@group/add-message/ADD_MESSAGE_REQUEST',
  '@group/add-message/ADD_MESSAGE_SUCCESS',
  '@group/add-message/ADD_MESSAGE_FAILURE'
)<{ id: string, groupId: string, body: string }, { id: string, groupId: string }, { id: string, groupId: string, error: any }>()
