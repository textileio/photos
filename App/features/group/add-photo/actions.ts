import { createAction } from 'typesafe-actions'
import { BlockInfo, Protobufs } from '@textile/react-native-sdk'

import { SharedImage, ProcessingImageError } from './models'

export const insertImage = createAction('processingImages/INSERT_IMAGE', (resolve) => {
  return (uuid: string, sharedImage: SharedImage, destinationThreadId: string, comment?: string) => resolve({ uuid, sharedImage, destinationThreadId, comment })
})

export const imagePrepared = createAction('processingImages/IMAGE_PREPARED', (resolve) => {
  return (uuid: string, preparedFiles: Protobufs.IMobilePreparedFiles) => resolve({ uuid, preparedFiles })
})

export const uploadStarted = createAction('processingImages/UPLOAD_STARTED', (resolve) => {
  return (uuid: string, uploadId: string) => resolve({ uuid, uploadId })
})

export const imageUploadProgress = createAction('processingImages/IMAGE_UPLOAD_PROGRESS', (resolve) => {
  return (uploadId: string, progress: number) => resolve({ uploadId, progress })
})

export const imageUploadComplete = createAction('processingImages/IMAGE_UPLOAD_COMPLETE', (resolve) => {
  return (uploadId: string, responseCode: string, responseBody: string) => resolve({ uploadId, responseCode, responseBody })
})

export const sharedToThread = createAction('processingImages/SHARED_TO_THREAD', (resolve) => {
  return (uuid: string, blockInfo: BlockInfo) => resolve({ uuid, blockInfo })
})

export const complete = createAction('processingImages/COMPLETE', (resolve) => {
  return (uuid: string) => resolve({ uuid })
})

export const retry = createAction('processingImages/RETRY', (resolve) => {
  return (uuid: string) => resolve({ uuid })
})

export const cancelRequest = createAction('processingImages/CANCEL', (resolve) => {
  return (uuid: string) => resolve({ uuid })
})

export const cancelComplete = createAction('processingImages/CANCEL_COMPLETE', (resolve) => {
  return (uuid: string) => resolve({ uuid })
})

export const error = createAction('processingImages/ERROR', (resolve) => {
  return (error: ProcessingImageError) => resolve({ error })
})
