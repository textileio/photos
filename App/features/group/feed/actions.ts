import { createAsyncAction } from 'typesafe-actions'
import { ThreadFeedItem } from '@textile/react-native-sdk'

export const refreshFeed = createAsyncAction(
  '@group/feed/REFRESH_FEED_REQUEST',
  '@group/feed/REFRESH_FEED_SUCCESS',
  '@group/feed/REFRESH_FEED_FAILURE'
)<{ id: string, limit?: number }, { id: string, items: ReadonlyArray<ThreadFeedItem> }, { id: string, error: any }>()

export const loadFeedItems = createAsyncAction(
  '@group/feed/LOAD_FEED_ITEMS_REQUEST',
  '@group/feed/LOAD_FEED_ITEMS_SUCCESS',
  '@group/feed/LOAD_FEED_ITEMS_FAILURE'
)<{ id: string, limit?: number }, { id: string, items: ReadonlyArray<ThreadFeedItem> }, { id: string, error: any}>()
