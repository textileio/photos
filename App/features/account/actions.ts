import { createAction, createAsyncAction } from 'typesafe-actions'
import { IContact } from '@textile/react-native-sdk'

import { SharedImage } from '../group/add-photo/models'

export const chooseProfilePhoto = createAsyncAction(
  'account/CHOOSE_PROFILE_PHOTO_REQUEST',
  'account/CHOOSE_PROFILE_PHOTO_SUCCESS',
  'account/CHOOSE_PROFILE_PHOTO_ERROR'
)<void, { image: SharedImage; data: string }, any>()

export const cancelProfilePhotoUpdate = createAction(
  'account/CANCEL_PROFILE_PHOTO_UPDATE',
  resolve => {
    return () => resolve()
  }
)

export const refreshProfileRequest = createAction(
  'account/REFRESH_PROFILE_REQUEST'
)

export const refreshProfileSuccess = createAction(
  'account/REFRESH_PROFILE_SUCCESS',
  resolve => {
    return (profile: IContact) => resolve({ profile })
  }
)

export const profileError = createAction('account/PROFILE_ERROR', resolve => {
  return (error: any) => resolve({ error })
})

export const refreshPeerIdRequest = createAction(
  'account/REFRESH_PEER_ID_REQUEST'
)

export const refreshPeerIdSuccess = createAction(
  'account/REFRESH_PEER_ID_SUCCESS',
  resolve => {
    return (peerId: string) => resolve({ peerId })
  }
)

export const refreshPeerIdError = createAction(
  'account/REFRESH_PEER_ID_ERROR',
  resolve => {
    return (error: any) => resolve({ error })
  }
)

export const refreshAddressRequest = createAction(
  'account/REFRESH_ADDRESS_REQUEST'
)

export const refreshAddressSuccess = createAction(
  'account/REFRESH_ADDRESS_SUCCESS',
  resolve => {
    return (address: string) => resolve({ address })
  }
)

export const refreshAddressError = createAction(
  'account/REFRESH_ADDRESS_ERROR',
  resolve => {
    return (error: any) => resolve({ error })
  }
)

export const setUsernameRequest = createAction(
  'account/SET_USERNAME_REQUEST',
  resolve => {
    return (username: string) => resolve({ username })
  }
)

export const setAvatar = createAsyncAction(
  'account/SET_AVATAR_REQUEST',
  'account/SET_AVATAR_SUCCESS',
  'account/SET_AVATAR_FAILURE'
)<SharedImage, void, any>()

export const setRecoveryPhrase = createAction(
  'account/SET_RECOVERY_PHRASE',
  resolve => {
    return (recoveryPhrase: string) => resolve({ recoveryPhrase })
  }
)
