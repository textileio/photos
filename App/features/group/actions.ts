import { feedActions as feed } from './feed'
import { addMessageActions as addMessage } from './add-message'
import { addPhotoActions as addPhoto } from './add-photo'

export { feed, addMessage, addPhoto }
