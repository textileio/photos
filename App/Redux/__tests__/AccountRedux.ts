import actions, { reducer } from '../AccountRedux'
import { pb } from '@textile/react-native-sdk'

const initialState = reducer(undefined, {} as any)
const profile: pb.IContact = {
  id: 'id',
  address: 'address',
  username: 'username',
  avatar: 'avatar',
  inboxes: [{
    peer: 'peer',
    address: 'address',
    api: 'api',
    protocol: 'protocol',
    node: 'node',
    url: 'url',
    swarm: ['swarm']
  }],
  threads: [],
  created: { seconds: 0, nanos: 0 },
  updated: { seconds: 0, nanos: 0 }
}
const peerId = 'peerId'
const error = 'error'
const avatar = 'avatar'
const recoveryPhrase = 'recoveryPhrase'

describe('account', () => {
  describe('initial state', () => {
    it('should match snapshot', () => {
      expect(initialState).toMatchSnapshot()
    })
  })
  describe('init', () => {
    it('should become initialized', () => {
      expect(initialState.initialized).toBe(false)
      const state = reducer(initialState, actions.initSuccess())
      expect(state.initialized).toBe(true)
    })
  })
  describe('profile', () => {
    it('should be processing from refresh', () => {
      const state0 =  reducer(initialState, actions.refreshProfileRequest())
      expect(state0.profile.processing).toBeTruthy()
    })
    it('should be processing from setUsername', () => {
      const state0 =  reducer(initialState, actions.setUsernameRequest('username'))
      expect(state0.profile.processing).toBeTruthy()
    })
    it('should update profile', () => {
      const state0 = reducer(initialState, actions.refreshProfileSuccess(profile))
      expect(state0.profile.value).toEqual(profile)
      expect(state0.profile.processing).toBeFalsy()
    })
    it('should track profile error', () => {
      const state0 = reducer(initialState, actions.profileError(error))
      expect(state0.profile.error).toEqual(error)
      expect(state0.profile.processing).toBeFalsy()
    })
  })
  describe('peer id', () => {
    it('should not change state for request', () => {
      const state0 =  reducer(initialState, actions.refreshPeerIdRequest())
      expect(state0).toEqual(initialState)
    })
    it('should update peer id', () => {
      const state0 = reducer(initialState, actions.refreshPeerIdSuccess(peerId))
      expect(state0.peerId.value).toEqual(peerId)
    })
    it('should track peer id error', () => {
      const state0 = reducer(initialState, actions.refreshPeerIdError(error))
      expect(state0.peerId.error).toEqual(error)
    })
  })
  describe('avatar', () => {
    it('should not change state for request', () => {
      const state0 = reducer(initialState, actions.setAvatarRequest('avatar'))
      expect(state0).toEqual(initialState)
    })
    it('should track avatar error', () => {
      const state0 = reducer(initialState, actions.setAvatarError(error))
      expect(state0.avatar.error).toEqual(error)
    })
    it('should track pending avatar', () => {
      const state0 = reducer(initialState, actions.setPendingAvatar(avatar))
      expect(state0.avatar.pending).toEqual(avatar)
    })
  })
  describe('recovery phrase', () => {
    it('should track revocery phrase', () => {
      const state0 = reducer(initialState, actions.setRecoveryPhrase(recoveryPhrase))
      expect(state0.recoveryPhrase).toEqual(recoveryPhrase)
    })
  })
})
