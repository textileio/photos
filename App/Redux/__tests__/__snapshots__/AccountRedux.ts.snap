// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`account initial state should match snapshot 1`] = `
Object {
  "address": Object {},
  "avatar": Object {},
  "cafeSessions": Object {
    "processing": false,
    "sessions": Array [],
  },
  "initialized": false,
  "peerId": Object {},
  "profile": Object {
    "processing": false,
  },
}
`;
